lib/utils/index.js

```diff
@@ -1778,9 +1778,11 @@ module.exports = {
 
   /**
    * @param {MemberExpression|Identifier} props
+   * @param {boolean} propProps
+   * @param {boolean} isRootProps
    * @returns { { kind: 'assignment' | 'update' | 'call' , node: ESNode, pathNodes: MemberExpression[] } | null }
    */
-  findMutating(props) {
+  findMutating(props, propProps = true, isRootProps = false) {
```

Comment:
The findMutating function is also used outside of rules related to props. So I don't want to add those options to this function.
By checking pathNodes in the rule implementation, I think you can determine if you are mutating properties.

Reply:
I can delete the param `isRootProps` and make it work, but it seems impossible to determine whether the options `shallowOnly(propProps)` is ture or false without modifying the `findMutating` function, because the return value of `target.type` `UpdateExpression` and `UnaryExpression` is totally same.

So, could I change the return value `kind` to `unary` when `target.type === 'UnaryExpression'`?
Or could I replace the param `propProps` with `options`?

Reply:
If UpdateExpression and UnaryExpression are of the same kind, I'm not sure what the problem is. Could you explain a little more?

Reply:
Oh, there is no problem, it's my fault. I have pushed a new commit to revert it.